### ISSUE TITLE
Nested HTML and Markdown doesn't render properly

### ISSUE THREAD

**Issue by sorachan** [notable/notable#1139]
- **OS Version**: Manjaro Linux (Arch-based, rolling)
- **Notable Version**: 1.8.4

### Current behavior

The program logic behind the `<markdown>` tag appears to be flawed. The following code is rendered as seen in the picture below.

```markdown
<details>
<summary>
Markdown, level 1
</summary>
<markdown>
*This* is Markdown. It renders properly.
<details>
<summary>
Markdown, level 2
</summary>
<markdown>
*This* is Markdown as well. It does not render properly in v1.8.4.
</markdown>
*This* should **not** be rendered as Markdown as it is inside an HTML tag, which works in v1.8.4.
</details>
*This*, however, is Markdown and should be rendered as such which isn't the case for v1.8.4.
</markdown>
</details>
```
![Screenshot_20200506_185136](https://user-images.githubusercontent.com/25408614/81205185-a1446000-8fca-11ea-93bc-decbfebcfbdc.png)

### Expected behavior

The `<markdown>` tag should work in a nested environment as described in the example above.

**Comment by sorachan** [comment]
Yet another instance of unexpected behaviour in the nested `<markdown>` tag:

```markdown
<details>
<summary>
Markdown, level 1
</summary>
<markdown>
...
<details>
<summary>
Markdown, level 2
</summary>
*This* should **not** be rendered as Markdown as it is inside an HTML tag, which works in v1.8.4.

**However**, *this* line is treated as Markdown when it should be treated as HTML as well...
</details>
...
</markdown>
</details>
```

![Screenshot_20200506_185746](https://user-images.githubusercontent.com/25408614/81205823-81616c00-8fcb-11ea-8403-f0a89565a60d.png)

**Comment by fabiospampinato** [comment]
I think I agree that this should work basically exactly as you want it to. 

Unfortunately right now there are multiple issues at play here.

- First of all HTML isn't parsed properly by our Markdown compiler, this is actually in accordance to the CommonMark spec, which in this scenario doesn't make sense to me. Basically in order to add a way to insert markdown in HTML they don't parse HTML properly, but with all sorts of weird rules, creating all sorts of problems, IMO a dedicated `<markdown>` tag would be better for this.
- Second the way Notable is parsing that code for implementing this `<markdown>` tag is also not with some proper HTML-parsing logic, but rather with a regex, which makes this brittle. In general you should indent your HTML properly too for it to work more reliably.
- Third I don't quite remember but in v1.8 this feature might not even support nested `<markdown>` tags.
- Lastly for some reason there are a bunch of `<br>` tags that v1.9 inserts that mess-up the layout (screenshot below for context).

Basically this feature is sort of broken right now, it might be getting somewhat reliable in v1.9 once the last issue I mentioned above gets fixed, when dealing with properly-indented HTML code, and it will most probably take a while before the other issues are addressed too, as they are fundamentally rooted in how CommonMark thinks Markdown should work, which we need to override.

![image](https://user-images.githubusercontent.com/1812093/81206424-034d9700-8fc4-11ea-80c6-80e05d53cd90.png)

**Comment by fabiospampinato** [comment]
You might be able to get the result to want via inserting Markdown in accordance to CommonMark's rules, which you should read the spec in order to understand better, but in essence the key thing is that if you leave an empty line after some HTML than the following will be parsed as Markdown, until another HTML tag is found and so on.
